### PR TITLE
feat(cli): Exit the test suite immediately upon `n` number of failing test suite

### PR DESCRIPTION
**Pull request by getspooky**
Alias: `-b`. Exit the test suite immediately upon `n` number of failing test suite. Minimum value is `1`.

Example :

```sh
deno test fetcher_test.ts --bail 3
```

Short style 

```
deno test fetcher_test.ts --b 3
```

### CODE REVIEW

**Comment by caspervonb** [comment]
We already have this with fail-fast?

**Comment by getspooky** [comment]
@caspervonb `fail-fast` will stop on first error but with `bail` we set ` n` number of failing test suite

**Comment by lucacasonato** [comment]
@getspooky What is the usecase for this? Does Go have this behaviour?

**Comment by getspooky** [comment]
By default, Deno runs all tests and throws all errors to the console upon completion. The `bail` config option can be used here to make Deno stop running tests after n failures. This options is inspired by [jest bail](https://jestjs.io/fr/docs/cli#--bail).

File: cli/tools/test_runner.rs [hunk 1]
```diff
@@ -532,6 +535,10 @@ pub async fn run_tests(
 
         reporter.visit_event(event);
 
+        if bail > 0 && failed > bail {
```
Comment (dsherret) [inline cli/tools/test_runner.rs:538]: Should this be `failed >= bail`?

Comment (getspooky) [reply cli/tools/test_runner.rs:538]: I think that number of failing test must be greater thant `0`.

Comment (dsherret) [reply cli/tools/test_runner.rs:538]: So the description says "Exit the test suite immediately upon n number of failing test suite". So, you would want this to exit when failed == bail, correct?

Anyway, I just doing a drive by review. Would be good to get consensus on this feature first.

**Comment by crowlKats** [comment]
if we are gonna pass this, it should be a single flag merged with `fail-fast`. we already have a plethora of flags and we have to be careful of adding new ones.

**Comment by getspooky** [comment]
If there is not need to add another flag like `--bail` , I think we should change the behaviour `fail-fast` to take `n` number and by default `Stop on first error`.
what do you think ?

**Comment by bartlomieju** [comment]
Haven't really seen such functionality, but I'd be okay to add optional integer argument to `--fail-fast` flag to give a threshold

**Comment by getspooky** [comment]
@bartlomieju  Yes it's common [use case](https://discord.com/channels/684898665143206084/684911491035430919/858143154943557632) , so we will still  change `--fail-fast` flag instread of using new one `--bail` ?

**Comment by bartlomieju** [comment]
> @bartlomieju  Yes it's common [use case](https://discord.com/channels/684898665143206084/684911491035430919/858143154943557632) , so we will still  change `--fail-fast` flag instread of using new one `--bail` ?

Yes

**Comment by bartlomieju** [comment]
Closing favor of #11159

**Pull request closed without merging**